File: packages/core/src/paths.ts
import { homedir } from "node:os";
import { resolve, join } from "node:path";

export interface AppPaths {
  home: string;
  config: string;
  repositories: string;
  mappings: string;
  workspaces: string;
  states: string;
  locks: string;
  logs: string;
}

export function getAppPaths(home = process.env.LOCAL_CONFIG_HOME ?? join(homedir(), ".local-config-sync")): AppPaths {
  const root = resolve(home);
  return {
    home: root,
    config: join(root, "config.yml"),
    repositories: join(root, "repositories.yml"),
    mappings: join(root, "mappings.yml"),
    workspaces: join(root, "workspaces"),
    states: join(root, "state", "repositories"),
    locks: join(root, "locks"),
    logs: join(root, "logs"),
  };
}

File: internal/core/paths.go
package core

import (
	"os"
	"path/filepath"
)

type AppPaths struct {
	Home         string
	Config       string
	Repositories string
	Mappings     string
	Workspaces   string
	States       string
	Locks        string
	Logs         string
}

func GetAppPaths(home string) AppPaths {
	if home == "" {
		home = os.Getenv("LOCAL_CONFIG_HOME")
	}
	if home == "" {
		userHome, _ := os.UserHomeDir()
		home = filepath.Join(userHome, ".local-config-sync")
	}
	home, _ = filepath.Abs(home)
	return AppPaths{
		Home: home, Config: filepath.Join(home, "config.yml"),
		Repositories: filepath.Join(home, "repositories.yml"),
		Mappings:     filepath.Join(home, "mappings.yml"), Workspaces: filepath.Join(home, "workspaces"),
		States: filepath.Join(home, "state", "repositories"), Locks: filepath.Join(home, "locks"),
		Logs: filepath.Join(home, "logs"),
	}
}
